server/src/services/person.service.ts

```diff
@@ -613,7 +621,8 @@ export class PersonService {
         }
 
         if (Object.keys(update).length > 0) {
-          primaryPerson = await this.repository.update({ id: primaryPerson.id, ...update });
+          const updated = await this.repository.update({ id: primaryPerson.id, ...update });
+          primaryPerson = updated[0];
```

Comment:
```suggestion
          const [primaryPerson] = await this.repository.update({ id: primaryPerson.id, ...update });
```
Similarly for the other cases.

Reply:
Fixed by https://github.com/immich-app/immich/pull/6455/commits/4bce98cfda869d5b72e71a4526f42e199c0c3975